### ISSUE TITLE
Misaligned pointer dereference in `gfx/platform/windows/font.rs`

### ISSUE THREAD

**Issue by dklassic** [servo/servo#31642]
I'm running servo through [verso](https://github.com/european-browser/verso) and currently a simple `cargo run` will result in panic:
```
>cargo run
   Compiling verso v0.0.1 (E:\GitHub\verso)
    Finished dev [unoptimized + debuginfo] target(s) in 12.98s
     Running `target\debug\verso.exe`
thread 'Style#0' panicked at C:\Users\[UserName]\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\gfx\platform\windows\font.rs:64:14:
misaligned pointer dereference: address must be a multiple of 0x4 but is 0x7ff75dec9a82
```

**Proposed Solution:**

Actually I have no idea what happened here and I'm not sure how to get it fixed, maybe the deref [here](https://github.com/servo/servo/blob/63527f56ca53ffad6be4d2552c8f7bb11bc945e2/components/gfx/platform/windows/font.rs#L64) is implemented wrong?

**Additional Details:**

- This issue will not show up if I run it in `--release` mode, which is extra confusing.
- Probably apparent but I'm running on Windows

**Comment by jdm** [comment]
What's the output with RUST_BACKTRACE=1?

**Comment by dklassic** [comment]
```
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1/library\std\src\panicking.rs:597
   1: core::panicking::panic_nounwind_fmt
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1/library\core\src\panicking.rs:106
   2: core::panicking::panic_misaligned_pointer_dereference
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1/library\core\src\panicking.rs:193
   3: gfx::platform::windows::font::make_tag
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\gfx\platform\windows\font.rs:64
   4: gfx::platform::windows::font::FontInfo::new_from_face
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\gfx\platform\windows\font.rs:96
   5: gfx::platform::windows::font::impl$6::new_from_template
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\gfx\platform\windows\font.rs:250
   6: gfx::font_context::FontContext<gfx::font_cache_thread::FontCacheThread>::create_font<gfx::font_cache_thread::FontCacheThread>
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\gfx\font_context.rs:211
   7: gfx::font_context::impl$0::get_font_maybe_synthesizing_small_caps::closure$0::closure$0<gfx::font_cache_thread::FontCacheThread>
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\gfx\font_context.rs:162
   8: enum2$<core::option::Option<gfx::font_cache_thread::FontTemplateInfo> >::and_then<gfx::font_cache_thread::FontTemplateInfo,gfx::font::Font,gfx::font_context::impl$0::get_font_maybe_synthesizing_small_caps::closure$0::closure_env$0<gfx::font_cache_thread::
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\core\src\option.rs:1414
   9: gfx::font_context::impl$0::get_font_maybe_synthesizing_small_caps::closure$0<gfx::font_cache_thread::FontCacheThread>
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\gfx\font_context.rs:159
  10: enum2$<core::option::Option<enum2$<core::option::Option<alloc::rc::Rc<core::cell::RefCell<gfx::font::Font>,alloc::alloc::Global> > > > >::unwrap_or_else<enum2$<core::option::Option<alloc::rc::Rc<core::cell::RefCell<gfx::font::Font>,alloc::alloc::Global> >
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\core\src\option.rs:979
  11: gfx::font_context::FontContext<gfx::font_cache_thread::FontCacheThread>::get_font_maybe_synthesizing_small_caps<gfx::font_cache_thread::FontCacheThread>
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\gfx\font_context.rs:153
  12: gfx::font_context::FontContext<gfx::font_cache_thread::FontCacheThread>::font<gfx::font_cache_thread::FontCacheThread>
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\gfx\font_context.rs:118
  13: gfx::font::FontGroupFamily::font<gfx::font_cache_thread::FontCacheThread>
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\gfx\font.rs:528
  14: gfx::font::impl$5::find::closure$0<gfx::font_cache_thread::FontCacheThread,gfx::font::impl$5::first::closure_env$0<gfx::font_cache_thread::FontCacheThread> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\gfx\font.rs:472
  15: core::iter::adapters::filter_map::filter_map_try_fold::closure$0<ref_mut$<gfx::font::FontGroupFamily>,alloc::rc::Rc<core::cell::RefCell<gfx::font::Font>,alloc::alloc::Global>,tuple$<>,enum2$<core::ops::control_flow::ControlFlow<alloc::rc::Rc<core::cell::R
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\core\src\iter\adapters\filter_map.rs:47
  16: core::iter::traits::iterator::Iterator::try_fold<core::slice::iter::IterMut<gfx::font::FontGroupFamily>,tuple$<>,core::iter::adapters::filter_map::filter_map_try_fold::closure_env$0<ref_mut$<gfx::font::FontGroupFamily>,alloc::rc::Rc<core::cell::RefCell<gf
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\core\src\iter\traits\iterator.rs:2461
  17: core::iter::adapters::filter_map::impl$2::try_fold<alloc::rc::Rc<core::cell::RefCell<gfx::font::Font>,alloc::alloc::Global>,core::slice::iter::IterMut<gfx::font::FontGroupFamily>,gfx::font::impl$5::find::closure_env$0<gfx::font_cache_thread::FontCacheThre
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\core\src\iter\adapters\filter_map.rs:137
  18: core::iter::traits::iterator::Iterator::find<core::iter::adapters::filter_map::FilterMap<core::slice::iter::IterMut<gfx::font::FontGroupFamily>,gfx::font::impl$5::find::closure_env$0<gfx::font_cache_thread::FontCacheThread,gfx::font::impl$5::first::closur
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\core\src\iter\traits\iterator.rs:2930
  19: gfx::font::FontGroup::find<gfx::font_cache_thread::FontCacheThread,gfx::font::impl$5::first::closure_env$0<gfx::font_cache_thread::FontCacheThread> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\gfx\font.rs:470
  20: gfx::font::FontGroup::first<gfx::font_cache_thread::FontCacheThread>
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\gfx\font.rs:459
  21: layout_2020::flow::text_run::TextRun::segment_text
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\text_run.rs:383
  22: layout_2020::flow::text_run::TextRun::break_and_shape
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\text_run.rs:217
  23: layout_2020::flow::inline::impl$5::break_and_shape_text::closure$0::closure$0
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\inline.rs:1681
  24: layout_2020::flow::inline::InlineFormattingContext::foreach<layout_2020::flow::inline::impl$5::break_and_shape_text::closure$0::closure_env$0>
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\inline.rs:1503
  25: layout_2020::flow::inline::impl$5::break_and_shape_text::closure$0
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\inline.rs:1677
  26: layout_2020::context::with_thread_local_font_context::closure$0<layout_2020::flow::inline::impl$5::break_and_shape_text::closure_env$0,tuple$<> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\context.rs:145
  27: std::thread::local::LocalKey<core::cell::RefCell<enum2$<core::option::Option<gfx::font_context::FontContext<gfx::font_cache_thread::FontCacheThread> > > > >::try_with<core::cell::RefCell<enum2$<core::option::Option<gfx::font_context::FontContext<gfx::font
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\std\src\thread\local.rs:270
  28: std::thread::local::LocalKey<core::cell::RefCell<enum2$<core::option::Option<gfx::font_context::FontContext<gfx::font_cache_thread::FontCacheThread> > > > >::with<core::cell::RefCell<enum2$<core::option::Option<gfx::font_context::FontContext<gfx::font_cac
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\std\src\thread\local.rs:246
  29: layout_2020::context::with_thread_local_font_context<layout_2020::flow::inline::impl$5::break_and_shape_text::closure_env$0,tuple$<> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\context.rs:144
  30: layout_2020::flow::inline::InlineFormattingContext::break_and_shape_text
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\inline.rs:1675
  31: enum2$<layout_2020::flow::BlockContainer>::construct_inline_formatting_context
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\construct.rs:224
  32: layout_2020::flow::construct::BlockContainerBuilder<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >::finish<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\construct.rs:261
  33: enum2$<layout_2020::flow::BlockContainer>::construct<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\construct.rs:215
  34: layout_2020::flow::BlockFormattingContext::construct<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\construct.rs:40
  35: enum2$<layout_2020::formatting_contexts::IndependentFormattingContext>::construct<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\formatting_contexts.rs:101
  36: layout_2020::flow::construct::BlockContainerBuilder<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >::handle_inline_level_element<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\construct.rs:497
  37: layout_2020::flow::construct::impl$3::handle_element<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\construct.rs:361
  38: layout_2020::dom_traversal::traverse_element<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData>,layout_2020::flow::construct::BlockContainerBuilder<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> > >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\dom_traversal.rs:197
  39: layout_2020::dom_traversal::traverse_children_of<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData>,layout_2020::flow::construct::BlockContainerBuilder<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> > >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\dom_traversal.rs:164
  40: enum2$<layout_2020::dom_traversal::NonReplacedContents>::traverse<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData>,layout_2020::flow::construct::BlockContainerBuilder<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DO
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\dom_traversal.rs:327
  41: enum2$<layout_2020::flow::BlockContainer>::construct<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\construct.rs:214
  42: enum2$<layout_2020::flow::construct::IntermediateBlockContainer>::finish<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\construct.rs:790
  43: layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >::finish<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\construct.rs:725
  44: layout_2020::flow::construct::impl$2::finish::closure$0<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\construct.rs:273
  45: core::ops::function::impls::impl$1::call_mut<tuple$<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> > >,layout_2020::flow::construct::impl$2::finish::closure_env$0<script::layout_dom::
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\core\src\ops\function.rs:272
  46: core::ops::function::impls::impl$4::call_once<tuple$<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> > >,ref$<layout_2020::flow::construct::impl$2::finish::closure_env$0<script::layout
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\core\src\ops\function.rs:305
  47: enum2$<core::option::Option<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> > > >::map
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\core\src\option.rs:1075
  48: core::iter::adapters::map::impl$2::next<layout_2020::cell::ArcRefCell<enum2$<layout_2020::flow::BlockLevelBox> >,rayon::vec::SliceDrain<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData>
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\core\src\iter\adapters\map.rs:103
  49: rayon::iter::plumbing::Folder::consume_iter<rayon::iter::collect::consumer::CollectResult<layout_2020::cell::ArcRefCell<enum2$<layout_2020::flow::BlockLevelBox> > >,layout_2020::cell::ArcRefCell<enum2$<layout_2020::flow::BlockLevelBox> >,core::iter::adapt
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\plumbing\mod.rs:178
  50: rayon::iter::map::impl$8::consume_iter<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >,layout_2020::cell::ArcRefCell<enum2$<layout_2020::flow::BlockLevelBox> >,rayon::iter::collect::
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\map.rs:248
  51: rayon::iter::plumbing::Producer::fold_with<rayon::vec::DrainProducer<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> > >,rayon::iter::map::MapFolder<rayon::iter::collect::consumer::Col
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\plumbing\mod.rs:110
  52: rayon::iter::plumbing::bridge_producer_consumer::helper<rayon::vec::DrainProducer<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> > >,rayon::iter::map::MapConsumer<rayon::iter::collect
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\plumbing\mod.rs:438
  53: rayon::iter::plumbing::bridge_producer_consumer<rayon::vec::DrainProducer<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> > >,rayon::iter::map::MapConsumer<rayon::iter::collect::consum
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\plumbing\mod.rs:397
  54: rayon::iter::plumbing::bridge::impl$0::callback<rayon::iter::map::MapConsumer<rayon::iter::collect::consumer::CollectConsumer<layout_2020::cell::ArcRefCell<enum2$<layout_2020::flow::BlockLevelBox> > >,layout_2020::flow::construct::impl$2::finish::closure_
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\plumbing\mod.rs:373
  55: rayon::vec::impl$7::with_producer<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >,rayon::iter::plumbing::bridge::Callback<rayon::iter::map::MapConsumer<rayon::iter::collect::consumer
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\vec.rs:147
  56: rayon::vec::impl$4::with_producer<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >,rayon::iter::plumbing::bridge::Callback<rayon::iter::map::MapConsumer<rayon::iter::collect::consumer
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\vec.rs:83
  57: rayon::iter::plumbing::bridge<rayon::vec::IntoIter<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> > >,rayon::iter::map::MapConsumer<rayon::iter::collect::consumer::CollectConsumer<lay
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\plumbing\mod.rs:357
  58: rayon::vec::impl$3::drive_unindexed<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >,rayon::iter::map::MapConsumer<rayon::iter::collect::consumer::CollectConsumer<layout_2020::cell::A
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\vec.rs:58
  59: rayon::iter::map::impl$2::drive_unindexed<rayon::vec::IntoIter<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> > >,layout_2020::flow::construct::impl$2::finish::closure_env$0<script::l
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\map.rs:49
  60: rayon::iter::collect::special_extend::closure$0<rayon::iter::map::Map<rayon::vec::IntoIter<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> > >,layout_2020::flow::construct::impl$2::fin
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\collect\mod.rs:39
  61: rayon::iter::collect::collect_with_consumer<layout_2020::cell::ArcRefCell<enum2$<layout_2020::flow::BlockLevelBox> >,rayon::iter::collect::special_extend::closure_env$0<rayon::iter::map::Map<rayon::vec::IntoIter<layout_2020::flow::construct::BlockLevelJob
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\collect\mod.rs:84
  62: rayon::iter::collect::special_extend<rayon::iter::map::Map<rayon::vec::IntoIter<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> > >,layout_2020::flow::construct::impl$2::finish::closur
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\collect\mod.rs:39
  63: rayon::iter::extend::impl$33::par_extend<layout_2020::cell::ArcRefCell<enum2$<layout_2020::flow::BlockLevelBox> >,rayon::iter::map::Map<rayon::vec::IntoIter<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020:
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\extend.rs:584
  64: rayon::iter::from_par_iter::collect_extended<alloc::vec::Vec<layout_2020::cell::ArcRefCell<enum2$<layout_2020::flow::BlockLevelBox> >,alloc::alloc::Global>,rayon::iter::map::Map<rayon::vec::IntoIter<layout_2020::flow::construct::BlockLevelJob<script::layo
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\from_par_iter.rs:20
  65: rayon::iter::from_par_iter::impl$0::from_par_iter<layout_2020::cell::ArcRefCell<enum2$<layout_2020::flow::BlockLevelBox> >,rayon::iter::map::Map<rayon::vec::IntoIter<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<lay
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\from_par_iter.rs:33
  66: rayon::iter::ParallelIterator::collect<rayon::iter::map::Map<rayon::vec::IntoIter<layout_2020::flow::construct::BlockLevelJob<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> > >,layout_2020::flow::construct::impl$2::finish::clos
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-1.9.0\src\iter\mod.rs:2069
  67: layout_2020::flow::construct::BlockContainerBuilder<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >::finish<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\construct.rs:271
  68: enum2$<layout_2020::flow::BlockContainer>::construct<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\construct.rs:215
  69: layout_2020::flow::BlockFormattingContext::construct<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\construct.rs:40
  70: enum2$<layout_2020::formatting_contexts::IndependentFormattingContext>::construct<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\formatting_contexts.rs:101
  71: layout_2020::flow::root::construct_for_root_element<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\root.rs:247
  72: layout_2020::flow::root::BoxTree::construct<script::layout_dom::node::ServoLayoutNode<layout_2020::dom::DOMLayoutData> >
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_2020\flow\root.rs:46
  73: layout_thread_2020::impl$9::handle_reflow::closure$2
             at C:\Users\DK\.cargo\git\checkouts\servo-12bfca331f128de3\d1fd8d5\components\layout_thread_2020\lib.rs:771
  74: core::ops::function::FnOnce::call_once<layout_thread_2020::impl$9::handle_reflow::closure_env$2,tuple$<> >
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\core\src\ops\function.rs:250
  75: rayon_core::thread_pool::impl$0::install::closure$0<layout_thread_2020::impl$9::handle_reflow::closure_env$2,tuple$<> >
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-core-1.12.1\src\thread_pool\mod.rs:147
  76: rayon_core::registry::impl$6::in_worker_cold::closure$0::closure$0<rayon_core::thread_pool::impl$0::install::closure_env$0<layout_thread_2020::impl$9::handle_reflow::closure_env$2,tuple$<> >,tuple$<> >
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-core-1.12.1\src\registry.rs:522
  77: rayon_core::job::impl$9::call::closure$0<tuple$<>,rayon_core::registry::impl$6::in_worker_cold::closure$0::closure_env$0<rayon_core::thread_pool::impl$0::install::closure_env$0<layout_thread_2020::impl$9::handle_reflow::closure_env$2,tuple$<> >,tuple$<> >
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-core-1.12.1\src\job.rs:218
  78: core::panic::unwind_safe::impl$23::call_once<tuple$<>,rayon_core::job::impl$9::call::closure_env$0<tuple$<>,rayon_core::registry::impl$6::in_worker_cold::closure$0::closure_env$0<rayon_core::thread_pool::impl$0::install::closure_env$0<layout_thread_2020::
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\core\src\panic\unwind_safe.rs:271
  79: std::panicking::try::do_call<core::panic::unwind_safe::AssertUnwindSafe<rayon_core::job::impl$9::call::closure_env$0<tuple$<>,rayon_core::registry::impl$6::in_worker_cold::closure$0::closure_env$0<rayon_core::thread_pool::impl$0::install::closure_env$0<la
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\std\src\panicking.rs:504
  80: layout_thread_2020::impl$15::deref
  81: std::panicking::try<tuple$<>,core::panic::unwind_safe::AssertUnwindSafe<rayon_core::job::impl$9::call::closure_env$0<tuple$<>,rayon_core::registry::impl$6::in_worker_cold::closure$0::closure_env$0<rayon_core::thread_pool::impl$0::install::closure_env$0<la
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\std\src\panicking.rs:468
  82: std::panic::catch_unwind<core::panic::unwind_safe::AssertUnwindSafe<rayon_core::job::impl$9::call::closure_env$0<tuple$<>,rayon_core::registry::impl$6::in_worker_cold::closure$0::closure_env$0<rayon_core::thread_pool::impl$0::install::closure_env$0<layout
             at /rustc/a28077b28a02b92985b3a3faecf92813155f1ea1\library\std\src\panic.rs:142
  83: rayon_core::unwind::halt_unwinding<rayon_core::job::impl$9::call::closure_env$0<tuple$<>,rayon_core::registry::impl$6::in_worker_cold::closure$0::closure_env$0<rayon_core::thread_pool::impl$0::install::closure_env$0<layout_thread_2020::impl$9::handle_refl
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-core-1.12.1\src\unwind.rs:17
  84: enum2$<rayon_core::job::JobResult<tuple$<> > >::call<tuple$<>,rayon_core::registry::impl$6::in_worker_cold::closure$0::closure_env$0<rayon_core::thread_pool::impl$0::install::closure_env$0<layout_thread_2020::impl$9::handle_reflow::closure_env$2,tuple$<>
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-core-1.12.1\src\job.rs:218
  85: rayon_core::job::impl$4::execute<rayon_core::latch::LatchRef<rayon_core::latch::LockLatch>,rayon_core::registry::impl$6::in_worker_cold::closure$0::closure_env$0<rayon_core::thread_pool::impl$0::install::closure_env$0<layout_thread_2020::impl$9::handle_re
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-core-1.12.1\src\job.rs:120
  86: rayon_core::job::JobRef::execute
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-core-1.12.1\src\job.rs:64
  87: rayon_core::registry::WorkerThread::execute
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-core-1.12.1\src\registry.rs:860
  88: rayon_core::registry::WorkerThread::wait_until_cold
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-core-1.12.1\src\registry.rs:794
  89: rayon_core::registry::WorkerThread::wait_until<rayon_core::latch::OnceLatch>
             at C:\Users\DK\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rayon-core-1.12.1\src\registry.rs:769
```

**Comment by mrobinson** [comment]
This is due to the implementation of `make_tag()` in `gfx/platform/windows/font.rs`. It should be packing the bytes into a u32 manually, but instead it is returning a pointer to a byte literal argument -- which must be stored at an unaligned memory locations (probably in the static data section).

**Comment by wusyong** [comment]
It seems #32127 also fixed this issue. `make_tag()` is renamed to `font_tag()` and it's now aligned well.

**Comment by mrobinson** [comment]
Yep! This was fixed by #32127.
